### PR TITLE
[ioctl] Add CandidateEndorsement and CandidateActivate action

### LINKED ISSUE #4191: add ioctl command for delegate endorsement
### What would you like to be added
add the following new commands in ioctl
- `ioctl stake2 activate`: activate a candidate with a bucket
- `ioctl stake2 endorse`: endorse a candidate with a bucket 
- `ioctl stake2 unendorse`: withdraw endorsement of a bucket

### Why is this needed
it's part of delegate endorsement feature

### How important you think this is for IoTeX
- [] must have
- [x] should have
- [] nice to have

### Additional information
Let us know any background or context that would help us better understand the request (for example the particular use-case that prompted this request)

### PR DESCRIPTION
**Pull request by envestcc**
# Description

Fixes #4191 

## Type of change
Please delete options that are not relevant.
- [x] Code refactor or improvement

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by CoderZhi — APPROVED** [review]

File: ioctl/cmd/action/stake2endorse.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	_stake2UnEndorseCmdUses = map[config.Language]string{
+		config.English: "unendorse BUCKET_INDEX" +
+			" [-s SIGNER] [-n NONCE] [-l GAS_LIMIT] [-p GAS_PRICE] [-P PASSWORD] [-y]",
+		config.Chinese: "unendorse 票索引" +
+			" [-s 签署人] [-n NONCE] [-l GAS 限制] [-p GAS 价格] [-P 密码] [-y]",
+	}
+
+	_stake2EndorseCmdShorts = map[config.Language]string{
+		config.English: "Endorse bucket's candidate on IoTeX blockchain",
+		config.Chinese: "在 IoTeX 区块链上给质押票的候选人背书",
```
Comment (dustinxie) [inline ioctl/cmd/action/stake2endorse.go:30]: nit:
在 IoTeX 区块链上背书候选人
在 IoTeX 区块链上撤销背书
this is more clear and straightforward?

File: ioctl/cmd/action/stake2register.go [hunk 2]
```diff
@@ -20,12 +20,12 @@ import (
 var (
 	_registerCmdUses = map[config.Language]string{
 		config.English: "register NAME (ALIAS|OPERATOR_ADDRESS) (ALIAS|REWARD_ADDRESS) (ALIAS|OWNER_ADDRESS) AMOUNT_IOTX STAKE_DURATION [DATA] [--auto-stake] [-s SIGNER] [-n NONCE] [-l GAS_LIMIT] [-p GAS_PRICE] [-P PASSWORD] [-y]",
-		config.Chinese: "register 名字 (别名|操作者地址）（别名|奖励地址）（别名|所有者地址）IOTX数量 质押持续时间 [数据] [--auto-stake] [-s 签署人] [-n NONCE] [-l GAS限制] [-p GAS价格] [-P 密码] [-y]",
+		config.Chinese: "register 名字 (别名 | 操作者地址）（别名 | 奖励地址）（别名 | 所有者地址）IOTX 数量 质押持续时间 [数据] [--auto-stake] [-s 签署人] [-n NONCE] [-l GAS 限制] [-p GAS 价格] [-P 密码] [-y]",
```
Comment (dustinxie) [inline ioctl/cmd/action/stake2register.go:23]: why need to do this?

File: ioctl/cmd/action/stake2register.go [hunk 3]
```diff
@@ -20,12 +20,12 @@ import (
...
 	}
 
 	_registerCmdShorts = map[config.Language]string{
 		config.English: "Register a candidate",
-		config.Chinese: "在IoTeX区块链上注册候选人",
+		config.Chinese: "在 IoTeX 区块链上注册候选人",
```
Comment (dustinxie) [inline ioctl/cmd/action/stake2register.go:28]: why need to do this?

File: ioctl/cmd/action/stake2endorse.go (hunk 1 above)
Comment (envestcc) [reply ioctl/cmd/action/stake2endorse.go:30]: okay

File: ioctl/cmd/action/stake2register.go (hunk 2 above)
Comment (envestcc) [reply ioctl/cmd/action/stake2register.go]: auto-formated by ide, reverted

File: ioctl/cmd/action/stake2register.go (hunk 3 above)
Comment (envestcc) [reply ioctl/cmd/action/stake2register.go]: reverted

File: ioctl/cmd/action/stake2endorse.go [hunk 4]
```diff
@@ -27,11 +27,11 @@ var (
 
 	_stake2EndorseCmdShorts = map[config.Language]string{
 		config.English: "Endorse bucket's candidate on IoTeX blockchain",
-		config.Chinese: "在 IoTeX 区块链上给质押票的候选人背书",
+		config.Chinese: "在 IoTeX 区块链上背书候选人",
```
Comment (dustinxie) [inline ioctl/cmd/action/stake2endorse.go:30]: remove 2 spaces

File: ioctl/cmd/action/stake2endorse.go [hunk 5]
```diff
@@ -27,11 +27,11 @@ var (
...
 	}
 	_stake2UnEndorseCmdShorts = map[config.Language]string{
 		config.English: "UnEndorse bucket's candidate on IoTeX blockchain",
-		config.Chinese: "在 IoTeX 区块链上给质押票的背书进行撤销",
+		config.Chinese: "在 IoTeX 区块链上撤销背书",
```
Comment (dustinxie) [inline ioctl/cmd/action/stake2endorse.go:34]: remove 2 spaces

**Review by dustinxie — APPROVED** [review]

**Pull request merged**